fenix/app/src/main/java/org/mozilla/fenix/components/appstate/AppState.kt

```diff
@@ -79,4 +83,22 @@ data class AppState(
     val wallpaperState: WallpaperState = WallpaperState.default,
     val standardSnackbarError: StandardSnackbarError? = null,
     val shoppingState: ShoppingState = ShoppingState(),
+    val screen: Screen = Screen.Home,
 ) : State
+
+sealed class Screen {
+
+    object Home : Screen()
+    data class Browser(
+        val searchTermOrURL: String,
+        val newTab: Boolean,
+        val from: BrowserDirection,
+        val customTabSessionId: String? = null,
+        val engine: SearchEngine? = null,
+        val forceSearch: Boolean = false,
+        val flags: EngineSession.LoadUrlFlags = EngineSession.LoadUrlFlags.none(),
+        val requestDesktopMode: Boolean = false,
+        val historyMetadata: HistoryMetadataKey? = null,
+        val additionalHeaders: Map<String, String>? = null,
+    ) : Screen()
```

Comment:
I expect that with more thought that some of this could be simplified or moved into a Store.

Reply:
I think this is a good idea. We should start with something ugly and then make it better iteratively.